File: limousine.php
<?php

class Limousine extends Vehicle
{
    // Define variables.
    private $HasMinibar;

    // Construct function.
    public function __construct($brand, $model, $buildYear, $licensePlate, $needsCleaning, $milage, $isAvailable, bool $hasMinibar)
    {
        parent::__construct($brand, $model, $buildYear, $licensePlate, $needsCleaning, $milage, $isAvailable);
        $this->HasMinibar = $hasMinibar;
    }

    // Price function.
    public function Price(float $distance)
    {
        if ($this->HasMinibar == true) {
            $price = 450 + $distance * 3 + 65;
        } else {
            $price = 450 + $distance * 3;
        }
        return round($price,2);   
    }

    // String function.
    public function __toString()
    {
        return
            parent::__toString() .
            "<br> HasMinibar: " . (boolval($this->HasMinibar) ? 'yes' : 'no');
    }
}

File: python/limousine.py
from dataclasses import dataclass
from vehicle import Vehicle

@dataclass
class Limousine(Vehicle):

    # Attributes.   
    minibar : bool

    # Calculate price function.   
    def Price(self, distance):
        if self.minibar == True:
            price = 450 + distance * 3 + 65
        else:
            price = 450 + distance * 3
        return round(price, 2)

    # To string function.
    def __str__(self):
        return f"{super().__str__()}\nMinibar: {self.minibar}"
    